src/gc.c

```diff
@@ -109,7 +110,10 @@ static void threadCallback(void* data) {
     RedisModule_ThreadSafeContextUnlock(ctx);
     return;
   }
-  gc->timerID = scheduleNext(gc);
+  // schedule next if not debug 
+  if (!bClient) {
```

Comment:
You need to schedule next if &gc->bClients is empty if you had a blocked client but now you do not have any more you still need to schedule, starting to think we really should put the blocked client as an argument to the task and remove gc->bClients

Reply:
@MeirShpilraien 
bClient is NULL only if the list is empty

Reply:
@ashtul it might be that this the last bc in this case you need to trigger the timmer but you will not and the GC will stop running...